examples/basic/client/src/main.rs

```diff
@@ -6,56 +6,73 @@ use com::{
 use interface::{Food, IAnimal, ICat, IDomesticAnimal, IExample, CLSID_CAT_CLASS};
 
 fn main() {
+    // Initialize the COM apartment
     init_apartment(ApartmentType::SingleThreaded)
         .unwrap_or_else(|hr| panic!("Failed to initialize COM Library{:x}", hr));
+    println!("Initialized apartment");
 
+    // Get a `BritishShortHairCat` class factory
     let factory = get_class_object::<IClassFactory>(&CLSID_CAT_CLASS)
         .unwrap_or_else(|hr| panic!("Failed to get cat class object 0x{:x}", hr));
     println!("Got cat class object");
 
+    // Get an instance of a `BritishShortHairCat` as the `IUnknown` interface
     let unknown = factory
         .get_instance::<IUnknown>()
         .expect("Failed to get IUnknown");
     println!("Got IUnknown");
 
+    // Now get a handle to the `IAnimal` interface
     let animal = unknown
         .get_interface::<IAnimal>()
```

Comment:
Why is `QueryInterface` renamed to `get_interface`?

Reply:
Same as above. This is the safe wrapper around `query_interface`. We could rename the unsafe version `QueryInterface` and the safe version could then be `query_interface`.

Reply:
Yes that may be better.